File: Python/ponto_encontro.py
from datetime import datetime, timedelta;

horario_encontro = datetime.fromisoformat(input()); # 2023-01-01T10:00:00Z
tempo_pessoa1 = int(input());      # -3 
fh_pessoa1 = int(input());         # 1 
tempo_pessoa2 = int(input());      # -2
fh_pessoa2 = int(input());         # 2

calc_pessoa1 = horario_encontro.hour - (fh_pessoa1 - (tempo_pessoa1));
saida_pessoa1 = horario_encontro.replace(hour=calc_pessoa1);

calc_pessoa2 = horario_encontro.hour - (fh_pessoa2 - (tempo_pessoa2));
saida_pessoa2 = horario_encontro.replace(hour=calc_pessoa2);

print("A primeira pessoa deve sair em " + str(saida_pessoa1)[0:19]);
print("A segunda pessoa deve sair em " + str(saida_pessoa2)[0:19]);

File: JS/pontoEncontro.js
import prompt from 'readline';
import util from 'util';
const rl = prompt.createInterface({ input: process.stdin, output: process.stdout });
const question = util.promisify(rl.question).bind(rl);
const read = (msg) => question(`${msg?msg+'\n':''}`);



let horarioEncontro = new Date(await read()); //2023-01-01T10:00:00Z
let tempoPessoa1 = await read();              //-3 
let fhPessoa1 = await read();                 //1 
let tempoPessoa2 = await read();              //-2
let fhPessoa2 = await read();                 //2

let calcPessoa1 = (horarioEncontro.getHours() + 3) - (fhPessoa1 - (tempoPessoa1));
let saidaPessoa1 = new Date(horarioEncontro);
saidaPessoa1.setHours(calcPessoa1 - 3);

let calcPessoa2 = (horarioEncontro.getHours() + 3) - (fhPessoa2 - (tempoPessoa2));
let saidaPessoa2 = new Date(horarioEncontro);
saidaPessoa2.setHours(calcPessoa2 - 3);

console.log("A primeira pessoa deve sair em " + saidaPessoa1.toISOString().replace("T", " ").substr(0, 19));
console.log("A segunda pessoa deve sair em " + saidaPessoa2.toISOString().replace("T", " ").substr(0, 19));
